File: frogment/src/main/java/ch/grze/frogment/frogment/AbstractFrogment.java
package ch.grze.frogment.frogment;

import android.support.v4.app.Fragment;

//still in Java due to lack of interop: https://youtrack.jetbrains.com/issue/KT-19415
public abstract class AbstractFrogment extends Fragment implements FrogmentInterface {
    private FrogmentComponent frogmentComponent;

    @Override
    public FrogmentComponent getFrogmentComponent() {
        return frogmentComponent;
    }

    @Override
    public void setFrogmentComponent(FrogmentComponent component) {
        frogmentComponent = component;
    }
}

File: frogment/src/main/java/ch/grze/frogment/frogment/AbstractFrogment.kt
package ch.grze.frogment.frogment

import android.support.v4.app.Fragment
import ch.grze.frogment.core.navigation.NavigatorAware

abstract class AbstractFrogment private constructor(override var frogmentComponent: FrogmentComponent) :
        Fragment(),
        FrogmentInterface,
        FrogmentDataAware by frogmentComponent,
        FrogmentActivityAware by frogmentComponent,
        NavigatorAware by frogmentComponent
{
    constructor() : this(FrogmentComponent())

    init {
        frogmentComponent.frogment = this
    }
}
